### PR TITLE
Rpc: Return error if block does not exist

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
The endpoints `getConfirmedBlock` and `getBlockTime` return success, but `null` data when a block is not available. This is particularly confusing in the latter case, since `getBlockTime` also returns `null` if the block time is not available.

#### Summary of Changes
Return a JSON-RPC error if block is not available for a slot

### CODE REVIEW

File: core/src/rpc_error.rs [hunk 1]
```diff
@@ -40,6 +44,11 @@ impl From<RpcCustomError> for Error {
                 message: "Transaction is not signed".to_string(),
                 data: None,
             },
+            RpcCustomError::BlockNotAvailable { slot } => Self {
+                code: ErrorCode::ServerError(JSON_RPC_SERVER_ERROR_4),
```
Comment (CriesofCarrots) [inline core/src/rpc_error.rs:48]: @mvines , in https://github.com/solana-labs/solana/pull/10630 this is `JSON_RPC_SERVER_ERROR_3`, but it wasn't clear to me why. Let me know if I should revert this change.

Comment (mvines) [reply core/src/rpc_error.rs:48]: I think that was just a silent protest against burning server errors.  We only have 100 of them, and then the world ends
